python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,457 @@
+        grouped into MultiPolygons.
+
+        As a result, a GpuGeoSeries object contains these arrow-supported
+        columns:
+        points_xy
+        mpoints_xy
+        mpoints_offsets
+        lines_xy
+        lines_offsets
+        lines_ml_offsets
+        polygons_xy
+        polygons_rings
+        polygons_polys
+        polygons_mpolys
+
+        Notes
+        -----
+        Legacy cuspatial algorithms depend on separated x and y columns.
+        Creating a cuGeoSeries from a GeoPandas source with `interleaved=True`
+        will create a legacy cuspatial Series, which violates the GeoArrow
+        format but will work with existing cuspatial algorithms.
+        """
+        self._data = data
+        self._reader = GeoSeriesReader(data, interleaved)
+        self._points = GpuPoints()
+        self._points.xy = self._reader.buffers[0]["points"]
+        self._multipoints = GpuMultiPoints()
+        self._multipoints.xy = self._reader.buffers[0]["multipoints"]
+        self._multipoints.offsets = self._reader.buffers[1]["multipoints"]
+        self._lines = GpuLines()
+        self._lines.xy = self._reader.buffers[0]["lines"]
+        self._lines.offsets = self._reader.buffers[1]["lines"]
+        self._lines.mlines = self._reader.buffers[1]["mlines"]
+        self._polygons = GpuPolygons()
+        self._polygons.xy = self._reader.buffers[0]["polygons"]["coords"]
+        self._polygons.polys = cudf.Series(
+            self._reader.offsets["polygons"]["polygons"]
+        )
+        self._polygons.rings = cudf.Series(
+            self._reader.offsets["polygons"]["rings"]
+        )
+        self._polygons.mpolys = cudf.Series(
+            self._reader.offsets["polygons"]["mpolys"]
+        )
+        self.types = self._reader.buffers[2]
+        self.lengths = self._reader.buffers[3]
+        self.index = cudf.Series(np.arange(len(self)))
+
+    class GeoSeriesLocIndexer:
+        def __init__(self):
+            raise NotImplementedError
+
+    class GeoSeriesILocIndexer:
+        def __init__(self, sr):
+            self._sr = sr
+
+        def __getitem__(self, index):
+            if not isinstance(index, slice):
+                return self._getitem_int(index)
+            else:
+                return self._getitem_slice(index)
+
+        def _getitem_int(self, index):
+            item_type = self._sr.types[index]
+            if item_type == "p":
+                return cuPoint(self._sr, index)
+            elif item_type == "mp":
+                return cuMultiPoint(self._sr, index)
+            elif item_type == "l":
+                return cuLineString(self._sr, index)
+            elif item_type == "ml":
+                return cuMultiLineString(self._sr, index)
+            elif item_type == "poly":
+                return cuPolygon(self._sr, index)
+            elif item_type == "mpoly":
+                return cuMultiPolygon(self._sr, index)
```

Comment:
IIRC using a dictionary mapping didn't work at all - this is because the constructors were all called during initialization on empty input arguments:
def map(argument):
    return ['mpoly': cuMultiPolygon(self._sr, index)][argument]
Crashed immediately.

Reply:
I mean something more like:
```python
_function_dict = {
    'p': cuPoint,
    'mp': cuMultiPoint,
    'l': cuLineString,
    'ml': cuMultiLineString,
    'poly': cuPolygon,
    'mpoly': cuMultiPolygon
}

def _getitem_int(self, index):
    return _function_dict[self._sr.types[index]](self._sr, index)
```

Reply:
Yeah, that's how I should have done it in the first place. :D Including `()` in a dict map calls the functions as they appear in the map, my bad.